CatManga search is broken

Any search on the CatManga source generates a `JSError TypeError` notification and no search results are shown.

Manga can still be added from the Discover and Library tabs - it seems to only be search that's broken..

<details>
<summary>Exported source logs (click to expand)</summary>

```
[INFO] [2021-09-12T04:57:59.481Z] [JavaScriptModel:init(for:mainFileLocation:):59] Injected Logger
[INFO] [2021-09-12T04:57:59.495Z] [SourceModel:injectTypes():38] Registered 31 Types
[INFO] [2021-09-12T04:57:59.497Z] [JavaScriptModel:init(for:mainFileLocation:):61] Injected Types
[INFO] [2021-09-12T04:57:59.501Z] [JavaScriptModel:injectTypeWrappers():112] Registered 40 Type Wrappers
[INFO] [2021-09-12T04:57:59.506Z] [JavaScriptModel:init(for:mainFileLocation:):63] Injected Type Wrappers
[INFO] [2021-09-12T04:57:59.510Z] [JavaScriptModel:init(for:mainFileLocation:):65] Injected App Constants
[INFO] [2021-09-12T04:57:59.642Z] [JavaScriptModel:initialiseExtension():125] Injected Cheerio.js - 0.22.0
[INFO] [2021-09-12T04:57:59.646Z] [JavaScriptModel:initialiseExtension():129] Injected source.CatManga.js
[INFO] [2021-09-12T04:57:59.647Z] [JavaScriptModel:init(for:mainFileLocation:):67] Initialised Extension: CatManga - 1.2.9 by PythonCoderAS
[ERROR] [2021-09-12T04:58:08.090Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[INFO] [2021-09-12T04:58:41.839Z] [JavaScriptModel:init(for:mainFileLocation:):59] Injected Logger
[INFO] [2021-09-12T04:58:41.851Z] [SourceModel:injectTypes():38] Registered 31 Types
[INFO] [2021-09-12T04:58:41.854Z] [JavaScriptModel:init(for:mainFileLocation:):61] Injected Types
[INFO] [2021-09-12T04:58:41.858Z] [JavaScriptModel:injectTypeWrappers():112] Registered 40 Type Wrappers
[INFO] [2021-09-12T04:58:41.859Z] [JavaScriptModel:init(for:mainFileLocation:):63] Injected Type Wrappers
[INFO] [2021-09-12T04:58:41.860Z] [JavaScriptModel:init(for:mainFileLocation:):65] Injected App Constants
[INFO] [2021-09-12T04:58:42.002Z] [JavaScriptModel:initialiseExtension():125] Injected Cheerio.js - 0.22.0
[INFO] [2021-09-12T04:58:42.008Z] [JavaScriptModel:initialiseExtension():129] Injected source.CatManga.js
[INFO] [2021-09-12T04:58:42.011Z] [JavaScriptModel:init(for:mainFileLocation:):67] Initialised Extension: CatManga - 1.2.9 by PythonCoderAS
[ERROR] [2021-09-12T04:58:50.931Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T04:59:23.801Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T04:59:25.260Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T04:59:25.981Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T04:59:26.722Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T04:59:47.860Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T04:59:48.758Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:09.623Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:10.714Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:13.116Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:28.697Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:30.102Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:33.868Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[ERROR] [2021-09-12T05:00:34.609Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[INFO] [2021-09-12T05:02:57.954Z] [JavaScriptModel:init(for:mainFileLocation:):59] Injected Logger
[INFO] [2021-09-12T05:02:57.966Z] [SourceModel:injectTypes():38] Registered 31 Types
[INFO] [2021-09-12T05:02:57.967Z] [JavaScriptModel:init(for:mainFileLocation:):61] Injected Types
[INFO] [2021-09-12T05:02:57.971Z] [JavaScriptModel:injectTypeWrappers():112] Registered 40 Type Wrappers
[INFO] [2021-09-12T05:02:57.974Z] [JavaScriptModel:init(for:mainFileLocation:):63] Injected Type Wrappers
[INFO] [2021-09-12T05:02:57.978Z] [JavaScriptModel:init(for:mainFileLocation:):65] Injected App Constants
[INFO] [2021-09-12T05:02:58.131Z] [JavaScriptModel:initialiseExtension():125] Injected Cheerio.js - 0.22.0
[INFO] [2021-09-12T05:02:58.134Z] [JavaScriptModel:initialiseExtension():129] Injected source.CatManga.js
[INFO] [2021-09-12T05:02:58.136Z] [JavaScriptModel:init(for:mainFileLocation:):67] Initialised Extension: CatManga - 1.2.9 by PythonCoderAS
[ERROR] [2021-09-12T05:03:02.614Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[INFO] [2021-09-12T06:06:59.178Z] [JavaScriptModel:init(for:mainFileLocation:):59] Injected Logger
[INFO] [2021-09-12T06:06:59.196Z] [SourceModel:injectTypes():38] Registered 31 Types
[INFO] [2021-09-12T06:06:59.200Z] [JavaScriptModel:init(for:mainFileLocation:):61] Injected Types
[INFO] [2021-09-12T06:06:59.207Z] [JavaScriptModel:injectTypeWrappers():112] Registered 40 Type Wrappers
[INFO] [2021-09-12T06:06:59.214Z] [JavaScriptModel:init(for:mainFileLocation:):63] Injected Type Wrappers
[INFO] [2021-09-12T06:06:59.220Z] [JavaScriptModel:init(for:mainFileLocation:):65] Injected App Constants
[INFO] [2021-09-12T06:06:59.374Z] [JavaScriptModel:initialiseExtension():125] Injected Cheerio.js - 0.22.0
[INFO] [2021-09-12T06:06:59.377Z] [JavaScriptModel:initialiseExtension():129] Injected source.CatManga.js
[INFO] [2021-09-12T06:06:59.379Z] [JavaScriptModel:init(for:mainFileLocation:):67] Initialised Extension: CatManga - 1.2.9 by PythonCoderAS
[ERROR] [2021-09-12T06:07:11.813Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
[INFO] [2021-09-12T07:22:38.916Z] [JavaScriptModel:init(for:mainFileLocation:):59] Injected Logger
[INFO] [2021-09-12T07:22:38.933Z] [SourceModel:injectTypes():38] Registered 31 Types
[INFO] [2021-09-12T07:22:38.937Z] [JavaScriptModel:init(for:mainFileLocation:):61] Injected Types
[INFO] [2021-09-12T07:22:38.947Z] [JavaScriptModel:injectTypeWrappers():112] Registered 40 Type Wrappers
[INFO] [2021-09-12T07:22:38.950Z] [JavaScriptModel:init(for:mainFileLocation:):63] Injected Type Wrappers
[INFO] [2021-09-12T07:22:38.952Z] [JavaScriptModel:init(for:mainFileLocation:):65] Injected App Constants
[INFO] [2021-09-12T07:22:39.175Z] [JavaScriptModel:initialiseExtension():125] Injected Cheerio.js - 0.22.0
[INFO] [2021-09-12T07:22:39.183Z] [JavaScriptModel:initialiseExtension():129] Injected source.CatManga.js
[INFO] [2021-09-12T07:22:39.186Z] [JavaScriptModel:init(for:mainFileLocation:):67] Initialised Extension: CatManga - 1.2.9 by PythonCoderAS
[ERROR] [2021-09-12T07:23:29.198Z] [SourceModelImplementation:searchResults(for:metadata:):31] JSError TypeError JSError TypeError
```
</details>

- Paperback v0.6.0-r2.0.13 (Public TestFlight)
- CatManga source v1.2.9

--- a/src/CatManga/CatManga.ts
+++ b/src/CatManga/CatManga.ts
@@ -15,7 +15,7 @@ const BASE = "https://catmanga.org"
 
 export const CatMangaInfo: SourceInfo = {
     icon: "icon.png",
-    version: "1.2.9",
+    version: "1.2.10",
     name: "CatManga",
     author: "PythonCoderAS",
     authorWebsite: "https://github.com/PythonCoderAS",
@@ -101,18 +101,15 @@ export class CatManga extends Source {
     async searchRequest(query: SearchRequest, metadata: any): Promise<PagedResults> {
         // TODO: Wait for search to be implemented on the website.
         const results = (await this.getWebsiteMangaDirectory(null)).results;
+
+        let data: MangaTile[];
         if (query.title){
-            query.title = query.title.replace(/\+/g, " ").trim();
+            const filterTitle = query.title.replace(/\+/g, " ").trim().toLowerCase();
+            data = results.filter((key) => (key.title.text || "").toLowerCase().includes(filterTitle))
+        } else {
+            data = results;
         }
-        const data: MangaTile[] = [];
-        for (let i = 0; i < results.length; i++) {
-            const key = results[i];
-            if (query.title) {
-                if ((key.title.text || "").toLowerCase().includes((query.title.toLowerCase()))) {
-                    data.push(key);
-                }
-            }
-        }
+
         console.log(data.length)
         return createPagedResults({
             results: data

